website/docs/developer/research/research-localstack.md

```diff
@@ -59,6 +59,323 @@ Docker's localhost can be accessed on the parent machine at http://host.docker.i
+}
+        
+        
+
+#### Lambda function
+`awslocal lambda create-function --function-name test-function --runtime python3.7 --zip-file fileb://test.zip --handler test.lambda_handler --role test`
+
+     Response: 
+    {
+        "FunctionName": "test-function",
+        "FunctionArn": "arn:aws:lambda:us-east-1:000000000000:function:test-function",
+        "Runtime": "python3.7",
+        "Role": "test",
+        "Handler": "test.lambda_handler",
+        "CodeSize": 217,
+        "Description": "",
+        "Timeout": 3,
+        "LastModified": "2021-06-16T18:32:27.753+0000",
+        "CodeSha256": "TxccBpGFdCdDKjbXygowwHMdUE0yPX+1U2eu6Vd+rsk=",
+        "Version": "$LATEST",
+        "VpcConfig": {},
+        "TracingConfig": {
+            "Mode": "PassThrough"
+        },
+        "RevisionId": "fb286270-3b68-4cae-9237-8efac356bee3",
+        "State": "Active",
+        "LastUpdateStatus": "Successful",
+        "PackageType": "Zip"
+    }
+#### SQS
+
+`awslocal sqs create-queue --queue-name test-queue`
+
+     Response: 
+    {
+        "QueueUrl": "http://localhost:4566/000000000000/test-queue"
+    }
+`awslocal sqs send-message --queue-url http://localhost:4566/000000000000/test-queue --message-body "test message." --delay-seconds 0`
+
+     Response: 
+    {
+      "MD5OfMessageBody": "5cbd04aaf0430ff7fac38ebd11b72083",
+      "MessageId": "0d4d8a75-276d-9687-6604-975cc0a84c68"
+  }
+`awslocal sqs receive-message --queue-url http://localhost:4566/000000000000/test-queue --attribute-names All --message-attribute-names All`
+
+     Response: 
+    {
+        "Messages": [
+            {
+                "MessageId": "0d4d8a75-276d-9687-6604-975cc0a84c68",
+                "ReceiptHandle": "ljwfjwrrlolejmrqcqbsrrxajygjsbkleesnyxtvwlffoikfstreipeutydtzhxosvjicazswakagyfhcxbrbpxxofkzsdbgjlofnpbmfxxtztrsmjkhkecfbqvdxnyotoujibxdavaiewtthscexnvplkmfrryisowurnxrymfzvyozubidbhsms",
+                "MD5OfBody": "5cbd04aaf0430ff7fac38ebd11b72083",
+                "Body": "test message.",
+                "Attributes": {
+                    "SenderId": "AIDAIT2UOQQY3AUEKVGXU",
+                    "SentTimestamp": "1623871655817",
+                    "ApproximateReceiveCount": "1",
+                    "ApproximateFirstReceiveTimestamp": "1623871770221"
+                }
+            }
+        ]
+    }
+
+
+#### Secerts manager
+`awslocal secretsmanager create-secret --name localstack-secret`
+
+      Response: 
+    {
+        "ARN": "arn:aws:secretsmanager:us-east-1:000000000000:secret:localstack-secret-PAnuzB",
+        "Name": "localstack-secret",
+        "VersionId": "8c6c77a1-6400-441a-b8f7-730abec4b9d5"
+    }
+
+
+### Deploying AWS resources using terraform and localstack
+
+Terraform can also be used to deploy and test AWS resources locally using localstack. Make sure to have Terraform installed on your local machine. An example of deploying services like lambda, API gateway, SQS, S3 and SNS is shown below:
+
```

Comment:
wrap the code below in a code block to maintain formatting.

Is there a good way we should update our terraform in the cumulus-orca repo to easily make the switch between real AWS and localstack for testing?

Reply:
to test in localstack, we should update the endpoints in "provider" section in our terraform to something like this:

        provider "aws" {
          region = "us-west-2"
          access_key = "123"
          secret_key = "xyz"
          skip_credentials_validation = true
          skip_requesting_account_id = true
          skip_metadata_api_check = true
          s3_force_path_style = true
          endpoints {
            s3 = "http://localhost:4566"
            sqs = "http://localhost:4566"
            apigateway = "http://localhost:4566"
            lambda = "http://localhost:4566"
            sns = "http://localhost:4566"
            secretsmanager = "http://localhost:4566"
            iam = "http://localhost:4566"
          }
        }